code/game/objects/items/stacks/stack.dm

```diff
@@ -250,12 +250,18 @@
 
 /obj/item/stack/proc/build_recipe_data(datum/stack_recipe/recipe)
 	var/list/data = list()
+	var/obj/result = recipe.result_type
 
 	data["uid"] = recipe.UID()
 	data["required_amount"] = recipe.req_amount
 	data["result_amount"] = recipe.res_amount
 	data["max_result_amount"] = recipe.max_res_amount
-	data["image"] = recipe.image
+	data["icon"] = result.icon
+	data["icon_state"] = result.icon_state
+
+	// DmIcon cannot paint images. So, if we have grayscale sprite, we need ready base64 image.
```

Comment:
Is this in reference to GAGS?

Reply:
Nah, TG haven't images on Stack Crafting UI, but we need to paint ~14 grayscale sprites for it
Anyway that's better than make all 300+ base64 sprites